### ISSUE TITLE
Problem when RSS domain differs from API domain - e.g. AltHub

### ISSUE THREAD

**Issue by PCurd** [evilhero/mylar#1776]
For whatever reason AltHub doesn't allow RSS requests on their API subdomain, you have to use the website domain. This means the RSS functions in Mylar don't work at all. Looking very quickly at the code I can see two ways this could be solved quickly (but I'm sure there are more!):

1) Create AltHub (and any others with this weirdness) in the same way as Dog and NZB.SU (as separate options) and then in rsscheck.nzbs set the RSS URL properly.
2) Create a magic string ending like [nzbhydra] and then in rsscheck.nzbs add another "if" check below the for [nzbhyrda] and override the site address

I suppose the "proper" solution is to add a setting to the search providers settings like "alternative RSS host" but that clutters up the interface.

Anyway, I'd appreciate any thoughts - I could send a PR for 2) but 1) requires a bit more digging and more than I'd risk doing on my first PR! If there is an option 3) that's not too scary I could tackle that too.

Thanks,

**Comment by evilhero** [comment]
Weird, that must have  just changed recently as I've done several tests/usage against ALTHub without any problems. I'm pretty sure you can still do an rss just call the most recent items from the given category (it's an api call) - as opposed to the rss call. This is what some other automation programs do - instead of using the rss directly, they call the api for the most recent ones.

I'll look into this now as I wasn't aware of this until you mentioned it - and it should be a pretty straightforward fix that I can just implement across all the newznab-based indexers...

**Comment by PCurd** [comment]
A forum thread from AltHub (from February 2017) which discusses:

https://althub.co.za/forums/viewtopic.php?f=25&t=291&hilit=rss&sid=e650ce3c9b2cbcd50b52fc08eed478d3

**Comment by evilhero** [comment]
Yep, saw that thread too ;)

I'm almost done the coding of it. Kept the newznab entries as RSS (since they don't count towards the api limit), but if the rss comes back with an error or the like, it will automatically try against the api and pull down the last 100 entries.

Also added in an auto-disable option, for when a site comes back unavailable it will automatically disable the site so it won't be used and then it will re-enable it again after 3 hours (or on a restart of Mylar, whichever occurs first). All that and some code cleanup - should see it in development tomorrow a.m. pending something blowing up prior.

**Comment by PCurd** [comment]
Forgive me if I'm misunderstanding (it's quite early here!) but does this change mean that if the RSS pull from the normal newznab host address fails you'll pull from the API to get the last 100 entries? So you're not actually using the RSS feed they do host? 

Either way, it'll still be better than what we have now!

Once I see the commit I'll pull it down and try it, thanks!

**Comment by evilhero** [comment]
It'll actually try to the normal RSS call so it goes against the RSS feed, and not the API. If the RSS call fails, it'll drop down to an API call and retrieve the last 100 items so as to ensure that the feed is continually populated in some manner.

It's only a problem for newznab sites that don't have an RSS option behind their api url (as is the case with ALTHub). A few other indexers I tried, it used the RSS option just fine and was hitting the site's rss url (I tested against 2 other sites). But with ALTHub, it will throw a warning saying unable to retrieve feed, dropping down to api - and then retrieve them just fine ;)
